polar-core/src/vm.rs

```diff
@@ -2932,14 +2952,13 @@ impl Runnable for PolarVirtualMachine {
         if let Some(value) = term {
             self.log_with(|| format!("=> {}", value.to_string()), &[]);
 
-            self.rebind_external_answer(
-                &self
-                    .call_id_symbols
-                    .get(&call_id)
-                    .expect("unregistered external call ID")
-                    .clone(),
-                value,
-            );
+            // Fetch variable to unify with call result.
+            let sym = &self.get_call_sym(&call_id).to_owned();
+
+            self.push_goal(Goal::Unify {
+                left: Term::new_temporary(Value::Variable(sym.to_owned())),
```

Comment:
if you remove the ref on 2956, you don't need the additional call to `.to_owned()` on 2959:
```suggestion
            let sym = self.get_call_sym(&call_id).to_owned();

            self.push_goal(Goal::Unify {
                left: Term::new_temporary(Value::Variable(sym)),
```